src/fondant/component/data_io.py

```diff
@@ -25,9 +27,11 @@ def __init__(
         manifest: Manifest,
         component_spec: ComponentSpec,
         input_partition_rows: t.Optional[int] = None,
+        consumes: t.Optional[t.Dict[str, str]] = None,
```

Comment:
I assume these are changes related to the pipeline interface change? Can we keep those out of this PR for now?